### PR TITLE
Rework to support StateObject

### PR DESCRIPTION
**Pull request by kean**
Designed according to https://developer.apple.com/videos/play/wwdc2020/10040/ recommendations.

### CODE REVIEW

**Comment by Sam-Spencer** [comment]
This looks great! I noticed in #7 that you expressed hesitation over how to proceed with this... 

Without realizing it -- in a private fork -- I think I independently came up with a similar solution to handle asynchronous Firebase URLs. I still have more testing to do, and I'd love to see what @StarLard has to say about this PR. *But* if it's any consolation, I think this looks like it'll work!

**Comment by kean** [comment]
> Without realizing it -- in a private fork -- I think I independently came up with a similar solution

Cool, yeah, I now think that's _the_ way to go. I initially was hesitant because of my lack of experience in using SwiftUI in my projects. I don't use FetchImage anywhere which also doesn't help.

**Comment by kean** [comment]
I'm also thinking I should remove lowData support for now. It makes things complicated and SwiftUI is already a nightmare to use. lowData mode support will be included in one of the upcoming Nuke releases.

File: README.md [hunk 1]
```diff
@@ -28,95 +30,55 @@ public final class FetchImage: ObservableObject, Identifiable {
...
 
-struct ImageView_Previews: PreviewProvider {
-    static var previews: some View {
-        let url = URL(string: "https://cloud.githubusercontent.com/assets/1567433/9781817/ecb16e82-57a0-11e5-9b43-6b4f52659997.jpg")!
-        return ImageView(image: FetchImage(url: url))
-            .frame(width: 80, height: 80)
-            .clipped()
+```swift
+struct DetailsView: View {
+    @State var refresh: Bool = false
```
Comment (joshuazmiller) [inline README.md:68]: Just pointing out this @State object is not used. Just a note for next time you update the readme

Comment (kean) [reply README.md:68]: Right, I accidentally pasted some test code. Thanks for pointing that out.

**Pull request merged**